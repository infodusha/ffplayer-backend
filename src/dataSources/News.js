import {query} from '../services/db.js';

/**
 * News data source
 */
export class News {
  /**
   * Get news
   * @param {number} offset
   * @param {number} length
   * @return {Promise<Array<any>>} news
   */
  list(offset, length) {
    return query('SELECT news_id AS id, title, text, date FROM news ORDER BY date DESC OFFSET $1 LIMIT $2', offset, length);
  }
}
