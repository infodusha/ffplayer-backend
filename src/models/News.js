import mongoose from 'mongoose';

const userSchema = new mongoose.Schema({
  title: {type: String, trim: true},
  text: {type: String, trim: true},
  date: {type: Date, default: Date.now},
});

export default mongoose.model('User', userSchema);
