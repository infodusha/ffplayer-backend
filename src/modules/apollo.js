import apolloM, * as apolloC from 'apollo-server-express';
const apollo = apolloM || apolloC;
export const gql = apollo.gql;
export const PubSub = apollo.PubSub;
export const ApolloError = apollo.ApolloError;
export const withFilter = apollo.withFilter;
export const SchemaDirectiveVisitor = apollo.SchemaDirectiveVisitor;
export const ApolloServer = apollo.ApolloServer;
