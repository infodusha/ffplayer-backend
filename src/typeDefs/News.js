import {gql} from '../modules/apollo.js';

export const News = gql`
  type News {
    id: ID!
    title: String!
    text: String
    date: Date!
  }
`;
