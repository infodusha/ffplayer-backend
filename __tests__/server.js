import {createTestClient} from 'apollo-server-testing';
import {ApolloServer, gql} from '../src/modules/apollo.js';
import {resolvers} from '../src/resolvers/index.js';
import {schemaDirectives} from '../src/schemaDirectives/index.js';
import {typeDefs} from '../src/typeDefs/index.js';
import {News} from '../src/dataSources/News.js';

describe('Server', () => {
  let client;
  beforeAll(() => {
    const news = new News();
    news.list = jest.fn().mockReturnValueOnce([
      {id: 1, date: new Date('10.11.2019')},
    ]);

    const context = () => ({user: {id: 1}, ip: '1.1.1.1'});

    const server = new ApolloServer({
      typeDefs,
      resolvers,
      schemaDirectives,
      context,
      dataSources: () => ({news}),
    });

    client = createTestClient(server);
  });

  it('Fetch news', async () => {
    const NEWS_QUERY = gql`query($length: Int!) {
      news(length: $length) {
        id
        date
      }
    }`;
    const res = await client.query({query: NEWS_QUERY, variables: {length: 1}});
    expect(res).toMatchSnapshot();
  });
});
