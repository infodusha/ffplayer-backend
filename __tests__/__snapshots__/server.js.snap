// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Fetch news 1`] = `
Object {
  "data": Object {
    "news": Array [
      Object {
        "date": 1570741200000,
        "id": "1",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
